### PR TITLE
cks: assorted fixes, test refactoring

### LINKED ISSUE #4265: CKS smoke test failures on master
Often, especially on Vmware and XenServer envs, the CKS smoke tests fail, for example:

test_02_deploy_kubernetes_ha_cluster | Error | 97.64 | test_kubernetes_clusters.py
test_04_deploy_and_upgrade_kubernetes_cluster | Error | 86.86 | test_kubernetes_clusters.py
test_05_deploy_and_upgrade_kubernetes_ha_cluster | Error | 0.04 | test_kubernetes_clusters.py
test_06_deploy_and_invalid_upgrade_kubernetes_cluster | Error | 72.59 | test_kubernetes_clusters.py
test_07_deploy_and_scale_kubernetes_cluster | Error | 70.56 | test_kubernetes_clusters.py
ContextSuite context=TestKubernetesCluster>:teardown | Error | 105.86 | test_kubernetes_clusters.py

Seen in https://github.com/apache/cloudstack/pull/4131#issuecomment-673851191

@shwstppr can you look into it next week, if there is some env tuning (timeout, retries etc) we should tune? Or, if tests themselves need improving to make their more reliable and deterministic?

### PR DESCRIPTION
**Pull request by shwstppr**
## Description
Fixes #4265 
Fixes for:
- template selection
- network cleanup
- capacity check
- specific patch version during deployment

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

File: tools/marvin/marvin/config/test_data.py [hunk 1]
```diff
@@ -1847,119 +1847,187 @@
...
+            "minmemory": 2048
+        },
+        "1.16.3": {
+            "semanticversion": "1.16.3",
+            "url": "http://download.cloudstack.org/cks/setup-1.16.3.iso",
+            "mincpunumber": 2,
+            "minmemory": 2048
         }
+    },
+    "cks_templates": {
```
Comment (rhtyd) [inline tools/marvin/marvin/config/test_data.py:1991]: @shwstppr you'll need to sync changes in test_data in trillian specific test_data as well

**Comment by rhtyd** [comment]
@shwstppr is this ready for testing?

File: tools/marvin/marvin/config/test_data.py (hunk 1 above)
Comment (shwstppr) [reply tools/marvin/marvin/config/test_data.py:1991]: Have done that

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
Changed base branch, moved master related regression fix in #4309

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@shwstppr is this in progress or ready for review/testing?
@blueorangutan package

**Comment by shwstppr** [comment]
@rhtyd changes are okay while deploying k8s clusters manually but Trillian run of smoke tests is taking over 2.5 hrs and double if there are failures in first run. I'm trying to reduce number of cluster deployments to reduce time

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@shwstppr unfortunately, tests are still intermittently failing

**Comment by rhtyd** [comment]
@davidjumani the pvlan test seems to be failing sometimes, can you help debug if it's env related or an intermittent failure?

**Comment by davidjumani** [comment]
@rhtyd Looks like it's because the same vlan id is used elsewhere. Will change it in another PR

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-67u3

**Review by rhtyd — APPROVED** [review]
LGTM, CKS tests passing

**Review by Pearl1594 — APPROVED** [review]
LGTM
Did some basic tests:
1. deployed a k8s cluster
2. destroyed the k8s cluster
3. Deleted the k8s template and re-registered another one with the same name as defined earlier (and as defined in the global setting) and verified that it picks up the new one and DOES NOT throw an error of being unable to find the template

**Pull request merged**